Orange/data/tests/test_variable.py

```diff
@@ -467,6 +471,37 @@ def test_make_proxy_cont(self):
         self.assertEqual(abc, abc1)
         self.assertEqual(abc, abc2)
         self.assertEqual(abc1, abc2)
+        self.assertEqual(hash(abc), hash(abc1))
+        self.assertEqual(hash(abc1), hash(abc2))
+
+    def test_proxy_has_separate_colors(self):
+        abc = ContinuousVariable("abc")
+        abc1 = abc.make_proxy()
+        abc2 = abc1.make_proxy()
+
+        original_colors = abc.colors
+        red_to_green = (255, 0, 0), (0, 255, 0), False
+        blue_to_red = (0, 0, 255), (255, 0, 0), False
+
+        abc1.colors = red_to_green
+        abc2.colors = blue_to_red
+        self.assertEqual(abc.colors, original_colors)
+        self.assertEqual(abc1.colors, red_to_green)
+        self.assertEqual(abc2.colors, blue_to_red)
+
+    def test_proxy_has_separate_attributes(self):
+        image = StringVariable("image")
+        image1 = image.make_proxy()
+        image2 = image1.make_proxy()
+
+        image.attributes["origin"] = "a"
+        image1.attributes["origin"] = "b"
+        image2.attributes["origin"] = "c"
+
+        self.assertEqual(image.attributes["origin"], "a")
+        self.assertEqual(image1.attributes["origin"], "b")
+        self.assertEqual(image2.attributes["origin"], "c")
```

Comment:
You could also add a simpler and more explicit test `self.assertIsNot(image.attributes, image1.attributes)`, and same for the other pairs.